src/kafkit/registry/manager.py

```diff
@@ -0,0 +1,312 @@
+"""Combined local and registry-based schema management."""
+
+from __future__ import annotations
+
+import json
+import logging
+from pathlib import Path
+from typing import TYPE_CHECKING, Any, Dict, Mapping, Optional
+
+from kafkit.registry.errors import RegistryBadRequestError
+from kafkit.registry.serializer import PolySerializer
+
+__all__ = ["RecordNameSchemaManager"]
+
+
+if TYPE_CHECKING:
+    from kafkit.registry.sansio import RegistryApi
```

Comment:
Small nit: I usually move the `if TYPE_CHECKING` block up before `__all__` so all the includes are together. `isort` doesn't put it there when it moves it, but it is happy with it there and won't move it again.